### PR TITLE
formatter: reject the condition if none

### LINKED ISSUE #1207: Treat None as a zero when using conditions
We can get this in `thunderbird_todos` where `priority` will be `None` if it's not used. Otherwise, it will be a number over `0`. Different number for `Low`, `Normal`, `High` too.
```
# display incompleted tasks with any priority
thunderbird_todos {
    format_todo = '\?if=!todo_completed [\?if=priority>0 {title}]'
    format_separator = ' '
}
```
I ran into this issue with `input` module too many moons ago.
```
format = '\?if=scroll_dy<0 down|up'    # scroll_dy is None
```
```
2017-09-30 03:39:03 WARNING Invalid format `[\?if=scroll_dy<0 down|up]` (TypeError) keyboard_locks.py line 293.
2017-09-30 03:39:03 INFO Traceback
TypeError: '<' not supported between instances of 'NoneType' and 'str'
  File "/home/chris/src/py3status/py3status/modules/keyboard_locks.py", line 293, in keyboard_mouse
    'full_text': self.py3.safe_format(self.format, self.x)
  File "/home/chris/src/py3status/py3status/py3.py", line 742, in safe_format
    attr_getter=attr_getter,
  File "/home/chris/src/py3status/py3status/formatter.py", line 213, in format
    valid, output = first_block.render(get_parameter, module)
  File "/home/chris/src/py3status/py3status/formatter.py", line 533, in render
    sub_valid, sub_output = item.render(get_params, module)
  File "/home/chris/src/py3status/py3status/formatter.py", line 522, in render
    valid = self.check_valid(get_params)
  File "/home/chris/src/py3status/py3status/formatter.py", line 505, in check_valid
    return self.commands._if.check_valid(get_params)
  File "/home/chris/src/py3status/py3status/formatter.py", line 367, in _check_valid_condition
    return (variable < value) == self.default
```

### PR DESCRIPTION
**Pull request by lasers**
Hi. I'm not familiar with the formatter code. This closes #1207 for me.
When we test a condition with `None`, an exception will just occur.
```python


        # This code block intentionally left blank.

```
We don't have anything in place for `None` inside the condition. The placeholder can be given `None` if it is not used or because it's not there (eg missing) or because it is not available initially due to threads or such. With this, we can return a `not self.default` value instead.

### CODE REVIEW

File: py3status/formatter.py [hunk 1]
```diff
@@ -345,8 +345,11 @@ def _check_valid_condition(self, get_params):
             variable = None
         value = self.value
 
+        # reject the condition?
+        if variable is None:
+            return not self.default
         # convert the value to a correct type
-        if isinstance(variable, bool):
+        elif isinstance(variable, bool):
```
Comment (tobes) [inline py3status/formatter.py]: Changing the `if` to `elif` does not make sense here as the proceeding `if`feels unrelated.  Also as we cannot get here if the proceeding condition is met.  So this just makes this PR harder to review.

Comment (lasers) [reply py3status/formatter.py]: I break them up. :+1:

File: py3status/formatter.py [hunk 2]
```diff
@@ -345,6 +345,10 @@ def _check_valid_condition(self, get_params):
             variable = None
         value = self.value
 
+        # reject the condition?
```
Comment (tobes) [inline py3status/formatter.py]: This comment does not help.  It spreads confusion.

There are many reasons for comments.  They are there to explain why we are doing things `reject the condition?` helps no-one.  Here we are dealing with a special case that you feel is a bug and worth of a PR to fix.  We should have an explanation of why we are doing this test.

But really it would help if we decide on our logic here. For example should `None=None`?

Comment (lasers) [reply py3status/formatter.py]: I'll write a different comment.

For me...
`None=None` is rejected if it's a None. Users should use `if=None` instead.
`None=None` is accepted if it's a string 'None`.

I thought about a condition for `is`, but I'm not sure if that's worth supporting. I have a branch for `==`, `>=`, `<=`, etc... as I don't like doing `if=volume>9` to get 10 or above.

**Comment by tobes** [comment]
thanks nice tests too :)
